MBBSEmu/Session/Telnet/TelnetSession.cs

```diff
@@ -41,13 +41,20 @@ private class TelnetOptionsValue {
 
         private readonly IacFilter _iacFilter;
 
-        public TelnetSession(ILogger logger, Socket telnetConnection) : base(logger, telnetConnection)
+        public TelnetSession(ILogger logger, Socket telnetConnection, AppSettings configuration) : base(logger, telnetConnection)
         {
             SessionType = EnumSessionType.Telnet;
             SessionState = EnumSessionState.Unauthenticated;
 
             _iacFilter = new IacFilter(logger);
             _iacFilter.IacVerbReceived += OnIacVerbReceived;
+
+            //Set Telnet Heartbeat from Config
+            _configuration = configuration;
+            if (!_configuration.TelnetHeartbeat)
```

Comment:
I would modify the overridden `Heartbeat` method to put the `if` statement there to not call the `base.Send` if this is set to false.

https://github.com/enusbaum/MBBSEmu/blob/e518a8b0251eea93c316d5f13f178e19dabb6ffe/MBBSEmu/Session/Telnet/TelnetSession.cs#L69